tools/clang/test/HLSLFileCheck/hlsl/intrinsics/wave/reduction/WaveBreakAndUnrollCS.hlsl

```diff
@@ -0,0 +1,41 @@
+// RUN: %dxc -T cs_6_0 %s | FileCheck %s
+// A test of explicit loop unrolling on a loop that uses a wave op in a break block
+
+// CHECK: void @main
+// CHECK: @dx.op.cbufferLoadLegacy.f32
+// CHECK: @dx.op.cbufferLoadLegacy.i32
+// CHECK: br i1
+// CHECK: @dx.op.cbufferLoadLegacy.f32
+// CHECK: @dx.op.cbufferLoadLegacy.i32
+// CHECK: br i1
+// CHECK: @dx.op.cbufferLoadLegacy.f32
+// CHECK: @dx.op.cbufferLoadLegacy.i32
+// CHECK: br i1
+// CHECK: @dx.op.cbufferLoadLegacy.f32
+// CHECK: @dx.op.cbufferLoadLegacy.i32
+// CHECK: br i1
+// CHECK: @dx.op.cbufferLoadLegacy.f32
+// CHECK: @dx.op.cbufferLoadLegacy.i32
+// CHECK: br i1
+// CHECK: @dx.op.cbufferLoadLegacy.f32
+// CHECK: br
+// CHECK: ret void
```

Comment:
Should we be checking that WaveActiveOp is repeated, rather than the cbuffer load, since that's the more important one?

And wouldn't this pass if unrolled without the ValueCache awareness, just with a lot more unrolled blocks?  I think we might need a `CHECK-NOT: WaveActiveOp` after checking for 5 unrolls, and before the check for `ret void`

Reply:
In my experience, it's been far easier to keep the wave operations where they should be inside the loop than the corresponding dependent buffer loads. However, the intent here is more about the loop unrolling, which that would accomplish just as well.